### PR TITLE
Separate envs

### PR DESCRIPTION
**Pull request by AungThiha**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -15,11 +15,90 @@ android {
...
+        }
+
+    }
+
+    compileOptions {
+        sourceCompatibility = 1.8
+        targetCompatibility = 1.8
+    }
+
+    dexOptions {
```
Comment (KhinThiriSoe) [inline app/build.gradle]: What does it use for?

File: app/build.gradle [hunk 2]
```diff
@@ -15,11 +15,90 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
 
+    signingConfigs {
```
Comment (KhinThiriSoe) [inline app/build.gradle]: What is signingConfigs?

File: base_library.gradle [hunk 3]
```diff
@@ -16,10 +16,34 @@ android {
...
-        release {
+
+        debug {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
+
+        release {
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```
Comment (KhinThiriSoe) [inline base_library.gradle]: proguardFiles here :D

File: core/src/dev/java/thiha/aung/boilerplate/core/config/ApiConfigs.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package thiha.aung.boilerplate.core.config
+
+const val PICSUM_API = "https://picsum.photos/"
+
+// often times, an app needs two apis
+// this is just a placeholder
+const val SECONDARY_API = "http://example.com/"
```
Comment (KhinThiriSoe) [inline core/src/dev/java/thiha/aung/boilerplate/core/config/ApiConfigs.kt]: http should always use https even if example?

File: app/build.gradle (hunk 2 above)
Comment (AungThiha) [reply app/build.gradle]: each object inside `signingConfigs` are signing configurations. They are used in `productFlavors.dev`, `productFlavors.staging` and `productFlavors.prod`.

This way, you don't need to manually import sign key and enter key password anymore. This is also helpful when using CI/CD

The configs are actually written inside `keystore.properties` and sign keys(keystores) should be stored inside `./keystores`

This is also explained in README

File: core/src/main/java/thiha/aung/boilerplate/core/data/remote/RemoteDataProperties.kt [hunk 5]
```diff
@@ -1,14 +1,4 @@
...
-    // used in photos module
-    const val PICSUM_VERSION_2_LIST = "v2/list"
-
-
-    // often times, an app needs two apis
-    // this is just a placeholder
-    const val SECONDARY_API = "http://example.com/"
-}
+// used in photos module. It's placed here so that we could all the routes easily
+const val PICSUM_VERSION_2_LIST = "v2/list"
```
Comment (KhinThiriSoe) [inline core/src/main/java/thiha/aung/boilerplate/core/data/remote/RemoteDataProperties.kt]: what about should have separated version?
PICSUM_VERSION = "v2"
PICSUM_LIST = "/list"

File: app/build.gradle (hunk 2 above)
Comment (AungThiha) [reply app/build.gradle]: you can see that at Line 20. It loads `keystore.properties`.

productFlavors starts from Line 73. Don't forget that all flavors need flavorDimensions.

File: app/build.gradle (hunk 1 above)
Comment (AungThiha) [reply app/build.gradle]: you can read about it here
https://stackoverflow.com/a/39932549/2353939

File: core/src/dev/java/thiha/aung/boilerplate/core/config/ApiConfigs.kt (hunk 4 above)
Comment (AungThiha) [reply core/src/dev/java/thiha/aung/boilerplate/core/config/ApiConfigs.kt]: Yes, you are right. We should always use secure network but this one is added just for an example. We don't really use it.

File: core/src/main/java/thiha/aung/boilerplate/core/data/remote/RemoteDataProperties.kt (hunk 5 above)
Comment (AungThiha) [reply core/src/main/java/thiha/aung/boilerplate/core/data/remote/RemoteDataProperties.kt]: with more variables, you are allocating more spaces on memory. It's better to reduce memory but of course, not at the expense of readability.

This one can be easily read and it's memory efficient. We need to separate it only when it's difficult to read.

File: base_library.gradle (hunk 3 above)
Comment (AungThiha) [reply base_library.gradle]: yes, it is there but I don't really understand what it is. haha
I'll read about it later.

Comment (AungThiha) [reply base_library.gradle]: Ok, now I know. Android studio now uses R8

File: core/src/main/java/thiha/aung/boilerplate/core/data/remote/RemoteDataProperties.kt (hunk 5 above)
Comment (AungThiha) [reply core/src/main/java/thiha/aung/boilerplate/core/data/remote/RemoteDataProperties.kt]: ah, we I guess it's better to put version 2 inside `PICSUM_API`. It's most likely going to be re-used in many places.

**Pull request merged**